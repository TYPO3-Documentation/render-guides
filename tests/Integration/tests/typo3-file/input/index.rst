===========
TYPO3 Files
===========

..  typo3:file:: ext_tables.sql
    :path: /
    :scope: extension
    :regex: /^.*ext_tables\.sql$/

    This file should contain a table-structure dump of the tables used by the
    extension which are not auto-generated.

..  typo3:file:: page.tsconfig
    :path: Configuration/
    :scope: extension
    :regex: ~^.*Configuration/page\.tsconfig$~

    In this file global page TSconfig can be stored. It will be automatically
    included for all pages.

..  typo3:file:: settings.php
    :composerPath: /config/system/
    :classicPath: /typo3conf/system/
    :scope: project
    :regex: /^.*settings\.php$/

    In this file global page TSconfig can be stored. It will be automatically
    included for all pages.

..  typo3:file:: ENABLE_INSTALL_TOOL
    :composerPath: /config/
    :classicPath: /typo3conf/
    :scope: project
    :language: plaintext
    :regex: /^.*ENABLE\_INSTALL\_TOOL$/

    When this file is set, it allows access to the TYPO3 Install Tool.

..  typo3:file:: LOCK_BACKEND
    :composerPath: /var/lock/
    :classicPath: /config/
    :scope: project
    :language: plaintext
    :configuration: :php:`$GLOBALS['TYPO3_CONF_VARS']['BE']['lockBackendFile']`
    :command: `vendor/bin/typo3 backend:lock`, `vendor/bin/typo3 backend:unlock`
    :regex: /^.*LOCK\_BACKEND$/

    When this file is set, it allows access to the TYPO3 Install Tool.

..  typo3:file:: config.yaml
    :scope: set
