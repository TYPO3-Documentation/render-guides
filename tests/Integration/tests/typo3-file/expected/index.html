<!-- content start -->
                <section class="section" id="typo3-files">
            <h1>TYPO3 Files<a class="headerlink" href="#typo3-files" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline"><i class="fa-solid fa-paragraph"></i></a></h1>
            <section data-search-title="ext_tables.sql"
         data-search-id="file-extension-ext-tables-sql"
         data-search-facet="file"
         class="confval-section"
>
    <h3 class="sr-only">ext_tables.sql</h3>
    <dl class="confval">
        <dt id="file-extension-ext-tables-sql"            class="d-flex justify-content-between">
            <div class="confval-header">
                <code class="sig-name descname"><span
                        class="pre">ext_<wbr>tables.<wbr>sql</span></code>
                                    <a class="headerlink" href="#file-extension-ext-tables-sql"
                       data-bs-toggle="modal"
                       data-bs-target="#linkReferenceModal"
                       data-id="file-extension-ext-tables-sql"
                       data-rstCode=":file:`packages/my_extension/ext_tables.sql`"
                       title="Reference this file definition"><i
                            class="fa-solid fa-paragraph"></i></a>
                </div>
        </dt>
        <dd>
            <dl class="field-list simple">
                                    <dt class="field-even">Scope</dt>
                    <dd class="field-even">extension</dd>
                                    <dt class="field-even">Path (Composer)</dt>
                    <dd class="field-even">packages/my_extension/ext_tables.sql</dd>
                                    <dt class="field-odd">Path (Classic)</dt>
                    <dd class="field-odd">typo3conf/ext/my_extension/ext_tables.sql</dd>
                </dl>
                        <div class="confval-description">

    <p>This file should contain a table-structure dump of the tables used by the
extension which are not auto-generated.</p>

            </div>
            </dd>
    </dl>
</section>
            <section data-search-title="Configuration/page.tsconfig"
         data-search-id="file-extension-configuration-page-tsconfig"
         data-search-facet="file"
         class="confval-section"
>
    <h3 class="sr-only">Configuration/page.tsconfig</h3>
    <dl class="confval">
        <dt id="file-extension-configuration-page-tsconfig"            class="d-flex justify-content-between">
            <div class="confval-header">
                <code class="sig-name descname"><span
                        class="pre">page.<wbr>tsconfig</span></code>
                                    <a class="headerlink" href="#file-extension-configuration-page-tsconfig"
                       data-bs-toggle="modal"
                       data-bs-target="#linkReferenceModal"
                       data-id="file-extension-configuration-page-tsconfig"
                       data-rstCode=":file:`packages/my_extension/page.tsconfig`"
                       title="Reference this file definition"><i
                            class="fa-solid fa-paragraph"></i></a>
                </div>
        </dt>
        <dd>
            <dl class="field-list simple">
                                    <dt class="field-even">Scope</dt>
                    <dd class="field-even">extension</dd>
                                    <dt class="field-even">Path</dt>
                    <dd class="field-even">Configuration/page.tsconfig</dd>
                                    <dt class="field-even">Path (Composer)</dt>
                    <dd class="field-even">packages/my_extension/Configuration/page.tsconfig</dd>
                                    <dt class="field-odd">Path (Classic)</dt>
                    <dd class="field-odd">typo3conf/ext/my_extension/Configuration/page.tsconfig</dd>
                </dl>
                        <div class="confval-description">

    <p>In this file global page TSconfig can be stored. It will be automatically
included for all pages.</p>

            </div>
            </dd>
    </dl>
</section>
            <section data-search-title="settings.php"
         data-search-id="file-project-config-system-settings-php"
         data-search-facet="file"
         class="confval-section"
>
    <h3 class="sr-only">settings.php</h3>
    <dl class="confval">
        <dt id="file-project-config-system-settings-php"            class="d-flex justify-content-between">
            <div class="confval-header">
                <code class="sig-name descname"><span
                        class="pre">settings.<wbr>php</span></code>
                                    <a class="headerlink" href="#file-project-config-system-settings-php"
                       data-bs-toggle="modal"
                       data-bs-target="#linkReferenceModal"
                       data-id="file-project-config-system-settings-php"
                       data-rstCode=":file:`config/system/settings.php`"
                       title="Reference this file definition"><i
                            class="fa-solid fa-paragraph"></i></a>
                </div>
        </dt>
        <dd>
            <dl class="field-list simple">
                                    <dt class="field-even">Scope</dt>
                    <dd class="field-even">project</dd>
                                    <dt class="field-even">Path (Composer)</dt>
                    <dd class="field-even">config/system/settings.php</dd>
                                    <dt class="field-odd">Path (Classic)</dt>
                    <dd class="field-odd">typo3conf/system/settings.php</dd>
                </dl>
                        <div class="confval-description">

    <p>In this file global page TSconfig can be stored. It will be automatically
included for all pages.</p>

            </div>
            </dd>
    </dl>
</section>
            <section data-search-title="ENABLE_INSTALL_TOOL"
         data-search-id="file-project-config-enable-install-tool"
         data-search-facet="file"
         class="confval-section"
>
    <h3 class="sr-only">ENABLE_INSTALL_TOOL</h3>
    <dl class="confval">
        <dt id="file-project-config-enable-install-tool"            class="d-flex justify-content-between">
            <div class="confval-header">
                <code class="sig-name descname"><span
                        class="pre">ENABLE_<wbr>INSTALL_<wbr>TOOL</span></code>
                                    <a class="headerlink" href="#file-project-config-enable-install-tool"
                       data-bs-toggle="modal"
                       data-bs-target="#linkReferenceModal"
                       data-id="file-project-config-enable-install-tool"
                       data-rstCode=":file:`config/ENABLE_INSTALL_TOOL`"
                       title="Reference this file definition"><i
                            class="fa-solid fa-paragraph"></i></a>
                </div>
        </dt>
        <dd>
            <dl class="field-list simple">
                                    <dt class="field-even">Scope</dt>
                    <dd class="field-even">project</dd>
                                    <dt class="field-even">Path (Composer)</dt>
                    <dd class="field-even">config/ENABLE_INSTALL_TOOL</dd>
                                    <dt class="field-odd">Path (Classic)</dt>
                    <dd class="field-odd">typo3conf/ENABLE_INSTALL_TOOL</dd>
                </dl>
                        <div class="confval-description">

    <p>When this file is set, it allows access to the TYPO3 Install Tool.</p>

            </div>
            </dd>
    </dl>
</section>
            <section data-search-title="LOCK_BACKEND"
         data-search-id="file-project-var-lock-lock-backend"
         data-search-facet="file"
         class="confval-section"
>
    <h3 class="sr-only">LOCK_BACKEND</h3>
    <dl class="confval">
        <dt id="file-project-var-lock-lock-backend"            class="d-flex justify-content-between">
            <div class="confval-header">
                <code class="sig-name descname"><span
                        class="pre">LOCK_<wbr>BACKEND</span></code>
                                    <a class="headerlink" href="#file-project-var-lock-lock-backend"
                       data-bs-toggle="modal"
                       data-bs-target="#linkReferenceModal"
                       data-id="file-project-var-lock-lock-backend"
                       data-rstCode=":file:`var/lock/LOCK_BACKEND`"
                       title="Reference this file definition"><i
                            class="fa-solid fa-paragraph"></i></a>
                </div>
        </dt>
        <dd>
            <dl class="field-list simple">
                                    <dt class="field-even">Scope</dt>
                    <dd class="field-even">project</dd>
                                    <dt class="field-even">Path (Composer)</dt>
                    <dd class="field-even">var/lock/LOCK_BACKEND</dd>
                                    <dt class="field-odd">Path (Classic)</dt>
                    <dd class="field-odd">config/LOCK_BACKEND</dd>
                                    <dt class="field-odd">Configuration</dt>
                    <dd class="field-odd">
    <p><code class="code-inline code-inline-long"
          translate="no"
          aria-description="Global PHP configuration"
          aria-details="The main configuration is achieved via a set of global
                settings stored in a global array called $GLOBALS['TYPO3_CONF_VARS'].
                They are commonly set in config/system/settings.php, config/system/additional.php,
                or the ext_localconf.php file of an extension. "
          data-bs-html="true"><a href="https://docs.typo3.org/m/typo3/reference-coreapi/13.4/en-us/Configuration/Typo3ConfVars/Index.html" target="_blank">$GLOBALS<wbr>[&#039;TYPO3_<wbr>CONF_<wbr>VARS&#039;]<wbr>[&#039;BE&#039;]<wbr>[&#039;lock<wbr>Backend<wbr>File&#039;] <span class="text-secondary"><i class="fa-regular fa-circle-question"></i></span></a></code></p>
</dd>
                                    <dt class="field-odd">Command</dt>
                    <dd class="field-odd">
    <p><code class="code-inline" translate="no">vendor/<wbr>bin/<wbr>typo3 backend:<wbr>lock</code>, <code class="code-inline" translate="no">vendor/<wbr>bin/<wbr>typo3 backend:<wbr>unlock</code></p>
</dd>
                </dl>
                        <div class="confval-description">

    <p>When this file is set, it allows access to the TYPO3 Install Tool.</p>

            </div>
            </dd>
    </dl>
</section>
            <section data-search-title="config.yaml"
         data-search-id="file-set-config-yaml"
         data-search-facet="file"
         class="confval-section"
>
    <h3 class="sr-only">config.yaml</h3>
    <dl class="confval">
        <dt id="file-set-config-yaml"            class="d-flex justify-content-between">
            <div class="confval-header">
                <code class="sig-name descname"><span
                        class="pre">config.<wbr>yaml</span></code>
                                    <a class="headerlink" href="#file-set-config-yaml"
                       data-bs-toggle="modal"
                       data-bs-target="#linkReferenceModal"
                       data-id="file-set-config-yaml"
                       data-rstCode=":file:`packages/my_extension/Configuration/Sets/MySet/config.yaml`"
                       title="Reference this file definition"><i
                            class="fa-solid fa-paragraph"></i></a>
                </div>
        </dt>
        <dd>
            <dl class="field-list simple">
                                    <dt class="field-even">Scope</dt>
                    <dd class="field-even">set</dd>
                                    <dt class="field-even">Path (Composer)</dt>
                    <dd class="field-even">packages/my_extension/Configuration/Sets/MySet/config.yaml</dd>
                                    <dt class="field-odd">Path (Classic)</dt>
                    <dd class="field-odd">typo3conf/ext/my_extension/Configuration/Sets/MySet/config.yaml</dd>
                </dl>
            </dd>
    </dl>
</section>
            <section data-search-title="SomeClass.php"
         data-search-id="file-extension-classes-someclass-php"
         data-search-facet="file"
         class="confval-section"
>
    <h3 class="sr-only">SomeClass.php</h3>
    <dl class="confval">
        <dt id="file-extension-classes-someclass-php"            class="d-flex justify-content-between">
            <div class="confval-header">
                <code class="sig-name descname"><span
                        class="pre">Some<wbr>Class.<wbr>php</span></code>
                                    <a class="headerlink" href="#file-extension-classes-someclass-php"
                       data-bs-toggle="modal"
                       data-bs-target="#linkReferenceModal"
                       data-id="file-extension-classes-someclass-php"
                       data-rstCode=":file:`packages/my_extension/Classes/SomeClass.php`"
                       title="Reference this file definition"><i
                            class="fa-solid fa-paragraph"></i></a>
                </div>
        </dt>
        <dd>
            <dl class="field-list simple">
                                    <dt class="field-even">Scope</dt>
                    <dd class="field-even">extension</dd>
                                    <dt class="field-even">Path (Composer)</dt>
                    <dd class="field-even">packages/my_extension/Classes/SomeClass.php</dd>
                                    <dt class="field-odd">Path (Classic)</dt>
                    <dd class="field-odd">typo3conf/ext/my_extension/Classes/SomeClass.php</dd>
                </dl>
            </dd>
    </dl>
</section>
            <section class="section" id="linking-files">
            <h2>Linking files<a class="headerlink" href="#linking-files" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline"><i class="fa-solid fa-paragraph"></i></a></h2>


<ul>
    <li><a class="composer-link"
           href="#file-project-config-enable-install-tool"
           title="File Information"
           data-bs-toggle="modal"
           data-bs-target="#generalModal"
           data-filename="ENABLE_INSTALL_TOOL"
           data-scope="project"
           data-classicpath="typo3conf/"
           data-composerpath="config/"
           data-classicpathprefix=""
           data-composerpathprefix=""
           data-shortdescription=""
        ><code class="code-inline file" translate="no">config/ENABLE_INSTALL_TOOL</code><i class="fa-regular fa-circle-question"></i></a></li>
    <li><a class="composer-link"
           href="#file-set-config-yaml"
           title="File Information"
           data-bs-toggle="modal"
           data-bs-target="#generalModal"
           data-filename="config.yaml"
           data-scope="set"
           data-classicpath=""
           data-composerpath=""
           data-classicpathprefix="typo3conf/ext/my_extension/Configuration/Sets/MySet/"
           data-composerpathprefix="packages/my_extension/Configuration/Sets/MySet/"
           data-shortdescription=""
        ><code class="code-inline file" translate="no">Configuration/Sets/MySet/config.yaml</code><i class="fa-regular fa-circle-question"></i></a></li>
    <li><a class="composer-link"
           href="#file-project-config-system-settings-php"
           title="File Information"
           data-bs-toggle="modal"
           data-bs-target="#generalModal"
           data-filename="settings.php"
           data-scope="project"
           data-classicpath="typo3conf/system/"
           data-composerpath="config/system/"
           data-classicpathprefix=""
           data-composerpathprefix=""
           data-shortdescription=""
        ><code class="code-inline file" translate="no">settings.php</code><i class="fa-regular fa-circle-question"></i></a></li>
    <li><a class="composer-link"
           href="#file-extension-ext-tables-sql"
           title="File Information"
           data-bs-toggle="modal"
           data-bs-target="#generalModal"
           data-filename="ext_tables.sql"
           data-scope="extension"
           data-classicpath=""
           data-composerpath=""
           data-classicpathprefix="typo3conf/ext/my_extension/"
           data-composerpathprefix="packages/my_extension/"
           data-shortdescription="Table-structure dump of the tables used by the extension"
        ><code class="code-inline file" translate="no">ext_tables.sql</code><i class="fa-regular fa-circle-question"></i></a></li>
    <li><a class="composer-link"
           href="#file-extension-classes-someclass-php"
           title="File Information"
           data-bs-toggle="modal"
           data-bs-target="#generalModal"
           data-filename="SomeClass.php"
           data-scope="extension"
           data-classicpath="Classes/"
           data-composerpath="Classes/"
           data-classicpathprefix="typo3conf/ext/my_extension/"
           data-composerpathprefix="packages/my_extension/"
           data-shortdescription="PHP Classes in this file get auto-loaded"
        ><code class="code-inline file" translate="no">Classes/SomeClass.php</code><i class="fa-regular fa-circle-question"></i></a></li>
    <li><a class="composer-link"
           href="#file-extension-classes-someclass-php"
           title="File Information"
           data-bs-toggle="modal"
           data-bs-target="#generalModal"
           data-filename="SomeClass.php"
           data-scope="extension"
           data-classicpath="Classes/"
           data-composerpath="Classes/"
           data-classicpathprefix="typo3conf/ext/my_extension/"
           data-composerpathprefix="packages/my_extension/"
           data-shortdescription="PHP Classes in this file get auto-loaded"
        ><code class="code-inline file" translate="no">EXT:my_extension/Classes/SomeClass.php</code><i class="fa-regular fa-circle-question"></i></a></li>
    <li><a class="composer-link"
           href="#file-set-config-yaml"
           title="File Information"
           data-bs-toggle="modal"
           data-bs-target="#generalModal"
           data-filename="config.yaml"
           data-scope="set"
           data-classicpath=""
           data-composerpath=""
           data-classicpathprefix="typo3conf/ext/my_extension/Configuration/Sets/MySet/"
           data-composerpathprefix="packages/my_extension/Configuration/Sets/MySet/"
           data-shortdescription=""
        ><code class="code-inline file" translate="no">Configuration File</code><i class="fa-regular fa-circle-question"></i></a></li>
    <li><code class="code-inline file" translate="no" title="File path">Unknown/<wbr>File.<wbr>xyz</code></li>
</ul>

    </section>
            <section class="section" id="code-block-captions">
            <h2>Code Block captions<a class="headerlink" href="#code-block-captions" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline"><i class="fa-solid fa-paragraph"></i></a></h2>
                <div class="code-block-caption">
        <span class="caption-text">Classes/SomeClass.php</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs php"><span class="hljs-keyword">echo</span> <span class="hljs-string">'Hello, TYPO3'</span>;</code></pre>
    <div class="code-block-actions">        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text"><code class="code-inline file" translate="no" title="File path">Classes/<wbr>Some<wbr>Class.<wbr>php</code></span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs php"><span class="hljs-keyword">echo</span> <span class="hljs-string">'Hello, TYPO3'</span>;</code></pre>
    <div class="code-block-actions">        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
    </section>
    </section>
        <!-- content end -->
